🚀 Sprint 3 - KPI Analysis
📖 Business Scenario

The CEO calls you into a meeting and says:

"Swetha, I don't have time to look at 100 reports. I need a dashboard that tells me how the business is performing."

He asks you to build Business KPIs.

learn:

What is a KPI?

How to calculate KPIs

How to interpret KPIs

How to explain KPIs

In [0]:
%sql
-- Total Revenue
select round(sum(`Sales`), 2) as Total_Revenue
from abc_retail_pvt_ltd.data_analytics.superstore

Total_Revenue
2326534.35


In [0]:
%sql
-- Total Profit
select round(sum(`Profit`), 2) as Total_Profit
from abc_retail_pvt_ltd.data_analytics.superstore;

Total_Profit
292296.81


📌 Task 3 - Profit Margin %

This is your first new KPI.

Formula:

Profit Margin =
(Total Profit / Total Revenue) × 100

Your task: Try writing the SQL yourself before asking me.

Hint: You'll need SUM(Profit) and SUM(Sales).

In [0]:
%sql
-- Profit Margin %
select round((sum(`Profit`) / sum(`Sales`) )*100, 2) as Profit_Margin
from abc_retail_pvt_ltd.data_analytics.superstore 


In [0]:
%sql
-- Total Orders
select count(*) as Total_Orders
from abc_retail_pvt_ltd.data_analytics.superstore;

In [0]:
%sql
-- Total Customers
select count(distinct `Customer ID`) as Total_Customers
from abc_retail_pvt_ltd.data_analytics.superstore;


📌 Task 6 - Average Order Value (AOV)
Business Question

On average, how much money does a customer spend in one order?

Formula:

### Total Revenue
### --------------
### Total Orders

In [0]:
%sql
-- Average Order Value (AOV)
select round(sum(`Sales`) / count(distinct `Order ID`),2) as AOV
from abc_retail_pvt_ltd.data_analytics.superstore;

📌 Task 7 - Average Profit Per Order

Formula:

### Total Profit
### -------------
### Total Orders

Try this yourself too.

In [0]:
%sql
-- Average Profit Per Order
select round(sum(`Profit`) / count(Distinct `Order ID`), 2) as APO
from abc_retail_pvt_ltd.data_analytics.superstore;

In [0]:
%sql
-- Average Discount
select round(avg(`Discount`)*100, 2) as Average_Discount
from abc_retail_pvt_ltd.data_analytics.superstore;

-- Notice we're multiplying by 100 because discounts are stored as decimals (e.g., 0.2 means 20%).

In [0]:
%sql
-- Total Quantity Sold
select sum(`Quantity`) as Total_items_sold
from abc_retail_pvt_ltd.data_analytics.superstore;

In [0]:
%sql
-- Sales per Customer
select round(sum(`Sales`)/ Count(distinct `Customer ID`),2) as Sales_per_customer
from abc_retail_pvt_ltd.data_analytics.superstore



KPI Values
 
Total Revenue          : $2,326,534.35

Total Profit           : $292,296.81

Profit Margin          : 12.56%

Average Order Value    : $455.20

Average Profit / Order : $57.18

Average Discount       : 15.54%

Sales per Customer     : $2,893.70



📊 KPI 1: Total Revenue

Value: $2,326,534.35

Business Interpretation

The company generated $2.33 million in revenue during the reporting period. This reflects the overall sales performance across all products, customers, and regions.

Business Recommendation

Continue focusing on high-performing product categories and regions while identifying opportunities to increase revenue in lower-performing markets.

📊 KPI 2: Total Profit

Value: $292,296.81

Business Interpretation

The company earned a total profit of $292,296.81, indicating that the business was profitable during the reporting period after accounting for costs reflected in the dataset.

Business Recommendation

Analyze products and regions with lower profitability to identify cost-saving opportunities and improve overall profit.

📊 KPI 3: Profit Margin

Value: 12.56%

Business Interpretation

The company retains approximately 12.56% of its revenue as profit. This means that for every $100 in sales, the business earns about $12.56 in profit.

Business Recommendation

Review pricing strategies and discount policies to improve the profit margin without negatively impacting sales.

📊 KPI 4: Average Order Value (AOV)

Value: $455.20

Business Interpretation

Each customer order generates an average revenue of $455.20, indicating the typical value of a single purchase.

Business Recommendation

Increase the average order value by introducing product bundles, cross-selling, or upselling strategies.

📊 KPI 5: Average Profit Per Order

Value: $57.18

Business Interpretation

On average, each order contributes $57.18 in profit, helping measure the profitability of individual transactions.

Business Recommendation

Focus on promoting high-margin products and reducing excessive discounts to improve profit per order.

📊 KPI 6: Average Discount

Value: 15.54%

Business Interpretation

Customers receive an average discount of 15.54% across all order lines. While discounts can encourage sales, they may also reduce profitability if not managed effectively.

Business Recommendation

Evaluate whether high discounts are driving sufficient additional sales to justify the reduction in profit.

📊 KPI 7: Sales per Customer

Value: $2,893.70

Business Interpretation

Each customer contributes approximately $2,893.70 in revenue on average, providing an indication of overall customer value.

Business Recommendation

Develop customer retention and loyalty programs to encourage repeat purchases and increase customer lifetime value.

⭐ This is how a Senior Analyst writes reports

Notice the pattern:

KPI  -->   Value  -->  Business Interpretation  -->  Business Recommendation

This is the structure followed in many analytics reports because it doesn't just present data—it explains what the data means and suggests what the business can do next.

## KPI: Total Revenue

**Value:** $2,326,534.35

### Business Interpretation
The company generated $2.33 million in revenue during the reporting period, indicating strong overall sales performance.

### Business Recommendation
Focus on expanding sales in high-performing regions while improving performance in underperforming markets.

%md

# 🧠 Data Analyst Thinking Process

Before writing any SQL query, always follow this approach.

```text
Business Requirement
        ↓
Understand the Business
        ↓
Write the Formula
        ↓
Think about the Dataset
        ↓
Write SQL
        ↓
Validate Results
        ↓
Business Insight
        ↓
Recommendation
```

---

## 📌 Step 1: Business Requirement
Understand what the stakeholder or business is asking.

**Example:**
> What is the Average Order Value (AOV)?

---

## 📌 Step 2: Understand the Business
Identify what the metric represents and why it is important.

**Example:**
Average Order Value tells us how much revenue is generated from a single order on average.

---

## 📌 Step 3: Write the Formula
Convert the business requirement into a mathematical formula.

**Example:**

Average Order Value = Total Sales / Total Orders

---

## 📌 Step 4: Think About the Dataset
Determine which columns are required and understand the grain of the data.

**Questions to ask:**
- Which column contains Sales?
- Which column identifies an Order?
- Does one row represent one order or one product within an order?
- Should I use `COUNT(*)` or `COUNT(DISTINCT Order ID)`?

---

## 📌 Step 5: Write SQL
Translate the formula into an SQL query.

---

## 📌 Step 6: Validate Results
Verify that the output is logical.

**Questions to ask:**
- Does the result make business sense?
- Are there any NULL values?
- Are duplicates affecting the calculation?
- Did I use the correct denominator?

---

## 📌 Step 7: Business Insight
Explain what the result means.

**Example:**
The average customer order generates **$455.20** in revenue.

---

## 📌 Step 8: Recommendation
Suggest an action based on the insight.

**Example:**
Increase the Average Order Value through cross-selling, upselling, or product bundle offers.

---

# 💡 Golden Rule

> **Never start with SQL.**
>
> Start with the business problem, understand the data, derive the formula, and then write SQL.
>
> A Data Analyst solves business problems—not just SQL queries.